tests/js/server/aql/aql-optimizer-rule-remove-collect-variables.js

```diff
@@ -158,9 +158,6 @@ function optimizerRuleTestSuite () {
                        "item3" : { "_id" : "ID" },
                      }
                    ],
-            "item1" : { "_id" : "ID" },
```

Comment:
So this test has actually been wrong?

Reply:
IMO, yes. It tests something totally different, so its not far-fetched that the author had a different focus and just tested against the current output.

But please have a look at the query to convince yourself.

Reply:
> COLLECT also provides an optional KEEP clause that can be used to control which variables will be copied into the variable created by INTO. If no KEEP clause is specified, all variables from the scope will be copied as sub-attributes into the groupsVariable. This is safe but can have a negative impact on performance if there are many variables in scope or the variables contain massive amounts of data.

Reply:
But note that
> The COLLECT statement will eliminate all local variables in the current scope. After COLLECT only the variables introduced by COLLECT itself are available.